modules/tensor_mechanics/include/bcs/Pressure.h

```diff
@@ -20,34 +21,40 @@ class Function;
 /**
  * Pressure applies a pressure on a given boundary in the direction defined by component
  */
-class Pressure : public IntegratedBC
+
+template <bool is_ad>
+using PressureParent = typename std::conditional<is_ad, ADIntegratedBC, IntegratedBC>::type;
+
+template <bool is_ad>
+class PressureTempl : public PressureParent<is_ad>
 {
 public:
   static InputParameters validParams();
 
-  Pressure(const InputParameters & parameters);
+  PressureTempl(const InputParameters & parameters);
 
 protected:
-  virtual void initialSetup() override;
-  virtual Real computeQpResidual() override;
-  virtual Real computeQpJacobian() override;
-  virtual void precalculateQpJacobian() override;
-  virtual Real computeQpOffDiagJacobian(const unsigned int jvar_num) override;
-  virtual void precalculateQpOffDiagJacobian(const MooseVariableFEBase & jvar) override;
+  virtual void initialSetup();
+  virtual GenericReal<is_ad> computeQpResidual();
+  virtual Real computeQpJacobian();
+  virtual void precalculateQpJacobian();
+  virtual Real computeQpOffDiagJacobian(const unsigned int jvar_num);
+  virtual void precalculateQpOffDiagJacobian(const MooseVariableFEBase & jvar);
```

Comment:
these should still override, they are not member function templates

Reply:
or is it not parsing it with the std::conditional

Reply:
afaik it s on a per routine basis

Reply:
The compiler will give a warning if you've used override once and could have used it again.  That's what I was trying to say.  But that doesn't work here because of different base classes.